enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/CdEnricher.java

```diff
@@ -74,7 +72,16 @@ protected boolean getDefaultOnline() {
         // when running in an Jenkins CD.
         // TODO: Isn't there a better datum to detect a CD build ? because BUILD_CD is set
         // also when you do a 'regular' CI job ....
-        return Systems.getEnvVarOrSystemProperty("BUILD_ID") != null;
+        String buildId = getBuildId();
+        return buildId != null;
+    }
+
+    private String getBuildId() {
```

Comment:
Is there any static class in charge of doing this inside FMP? I think this logic should be always in a static class so all properties behave with the same approach, and if someday it is decided to change, it is changed everywhere not just in the places we can remember to change.